framework/src/samplers/Sampler.C

```diff
@@ -0,0 +1,119 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "Sampler.h"
+#include "Distribution.h"
+#include <limits>
+
+template <>
+InputParameters
+validParams<Sampler>()
+{
+  InputParameters params = validParams<MooseObject>();
+  params += validParams<RandomInterface>();
+  params.addParam<unsigned int>(
+      "max", std::numeric_limits<unsigned int>::max(), "Maximum number of samples");
+  params.addParam<bool>("reseed", false, "Reseed for each new sample if this value is true");
+  params.addParam<std::vector<std::string>>("perturb_parameters",
+                                            std::vector<std::string>(),
+                                            "The names of the parameters that you want to perturb");
+  params.addParam<std::vector<std::string>>(
+      "distributions",
+      std::vector<std::string>(),
+      "The names of distributions that you want to use to perturb the given parameters");
+  params.registerBase("Sampler");
+  return params;
+}
+
+Sampler::Sampler(const InputParameters & parameters)
+  : MooseObject(parameters),
+    RandomInterface(parameters,
+                    *parameters.get<FEProblemBase *>("_fe_problem_base"),
+                    parameters.get<THREAD_ID>("_tid"),
+                    false),
+    DistributionInterface(this),
+    Restartable(parameters, "Samplers"),
+    _tid(getParam<THREAD_ID>("_tid")),
+    _max_samples(getParam<unsigned int>("max")),
```

Comment:
This does not seem to get used yet. What would limit the actual number of samples taken?